src/middlewared/middlewared/plugins/failover_/disabled_reasons.py

```diff
@@ -78,6 +79,9 @@ def get_local_reasons(self, app, ifaces, reasons):
         if self.middleware.call_sync('failover.config')['disabled']:
             reasons.add(DisabledReasonsEnum.NO_FAILOVER.name)
 
+        if self.middleware.call_sync('failover.reboot_required')['reboot_required']:
```

Comment:
`failover.reboot_required['reboot_required']` is redundant and is similar to what you did with the `truecommand.connected` endpoint. It's returning too much information for how it's named. We need to have an endpoint named `failover.reboot_info` that returns this dict of information. If we add `failover.reboot_required`, then we're going to have to consider returning only a boolean and only return true if the local node needs a reboot. If someone wants to know if the remote node needs to be rebooted, they'll have to call `failover.reboot_info`